packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,371 @@
+import { TransactionCall, TransactionWithSender, Web3BaseProvider } from 'web3-common';
+import { Web3Config, Web3RequestManager } from 'web3-core';
+import {
+	BlockNumberOrTag,
+	ValidTypes,
+	ValidReturnTypes,
+	convertToValidType,
+	Address,
+	Uint256,
+	HexString32Bytes,
+	HexStringBytes,
+	Uint,
+	HexString8Bytes,
+	convertObjectPropertiesToValidType,
+	isHexString32Bytes,
+	Filter,
+} from 'web3-utils';
+import {
+	convertibleBlockProperties,
+	convertibleFeeHistoryResultProperties,
+	convertibleReceiptInfoProperties,
+	convertibleTransactionInfoProperties,
+} from './convertible_properties';
+
+import * as RpcMethods from './rpc_methods';
+
+export class Web3Eth {
+	private readonly _requestManager: Web3RequestManager;
+	private readonly _options: Web3Config;
+
+	public constructor(provider: Web3BaseProvider | string, options: Web3Config) {
+		this._requestManager = new Web3RequestManager(provider);
+		this._options = options;
+	}
+
+	public async getProtocolVersion() {
+		return RpcMethods.getProtocolVersion(this._requestManager);
+	}
+
+	public async isSyncing() {
+		return RpcMethods.getSyncing(this._requestManager);
+	}
+
+	public async getCoinbase() {
+		return RpcMethods.getCoinbase(this._requestManager);
+	}
+
+	public async isMining() {
+		return RpcMethods.getMining(this._requestManager);
+	}
+
+	public async getHashRate<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getHashRate(this._requestManager);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async getGasPrice<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getGasPrice(this._requestManager);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async getAccounts() {
+		return RpcMethods.getAccounts(this._requestManager);
+	}
+
+	public async getBlockNumber<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getBlockNumber(this._requestManager);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	// TODO Discuss the use of multiple optional parameters
+	public async getBalance<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		address: Address,
+		blockNumber: BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getBalance(this._requestManager, address, blockNumber);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
```

Comment:
Wanted to get everyone's input on the use of multiple optional parameters. For this method, if a user wanted to default `blockNumber`, but wanted to specify `returnType`, they'd need to call the methods like so:

```typescript
getBalance('0x0...', undefined, ReturnType.BigInt);
```

I'm not sure how else to do this, just wanted to point it out because this is something we'll have to make clear in our documentation

Reply:
Is that how 1.x deals with multiple optional parameters?

Reply:
> Is that how 1.x deals with multiple optional parameters?

@luu-alex No, `1.x` uses the `...arguements` approach and doesn't specify each parameter in the function signature like we are doing now with TypeScript